python_modules/dagster/dagster/core/asset_defs/decorators.py

```diff
@@ -217,13 +231,18 @@ def build_asset_ins(
             namespace = None
             dagster_type = None
 
-        asset_key = AssetKey(list(filter(None, [namespace or asset_namespace, input_name])))
+        asset_key = asset_key or AssetKey(
+            list(filter(None, [namespace or asset_namespace, input_name]))
+        )
 
-        ins[asset_key] = In(
+        ins[input_name] = In(
             metadata=metadata,
             root_manager_key="root_manager",
             asset_key=asset_key,
             dagster_type=dagster_type,
         )
 
+    for asset_key in depends_on:
+        ins[asset_key.to_string(legacy=True)] = In(dagster_type=Nothing, asset_key=asset_key)
```

Comment:
Not sure what the best key to provide to the `ins` dict is. In order to enforce uniqueness, I set the stringified asset key (separated by `.` delimiter) to be the key. Though it probably doesn't matter because that key in `ins` would never be used in the op function?

Reply:
Missing some context here, but what was the issue with using the AssetKey as the key to this dictionary?

Reply:
the dictionary is passed to the ops as `ins`, so all the keys must be strings